libs/log/default.go

```diff
@@ -26,7 +30,11 @@ type defaultLogger struct {
 // Since zerolog supports typed structured logging and it is difficult to reflect
 // that in a generic interface, all logging methods accept a series of key/value
 // pair tuples, where the key must be a string.
-func NewDefaultLogger(format, level string, trace bool) (Logger, error) {
+func NewDefaultLogger(format, level string, trace bool, elementLength int) (Logger, error) {
+	if elementLength < 0 || elementLength > MaxLogElementLength {
+		return nil, fmt.Errorf("unsupported log element length: %d", elementLength)
+	}
+
```

Comment:
Since logging happens a lot, I suggest precompiling the formatter. That also saves some duplication in the code below:

```suggestion

	var format func(interface{}) string
	if elementLength > 0 {
		format = func(v interface{}) string {
			s := fmt.Sprint(v)
			if len(s) > elementLength {
				return s[:elementLength]
			}
			return s
		}
	} else {
		format = func(v interface{}) string {
			return fmt.Sprint(v)
		}
	}
```